experimental/auth-logic/wrappers/rekor_wrapper.go

```diff
@@ -101,31 +104,32 @@ func getRekordEntryFromAnon(logEntryAnon models.LogEntryAnon) (*rekord.V001Entry
 
 // Verify signature in a rekord entry. In the context where this is used,
 // this will verify the contents of a rekord entry (an endorsement file)
-// against the product team's public key.
-func verifyRekordLogSignature(rekordEntry *rekord.V001Entry) error {
+// against the product team's public key. It returns the public key if one can
+// be parsed from the entry.
+func verifyRekordLogSignature(rekordEntry *rekord.V001Entry) (*ecdsa.PublicKey, error) {
 	publicKey := rekordEntry.RekordObj.Signature.PublicKey.Content
 	// The unused argument is for extra bytes, not an error
 	block, _ := pem.Decode(publicKey)
 	pubKeyDecoded, err := x509.ParsePKIXPublicKey(block.Bytes)
 	if err != nil {
-		return fmt.Errorf("could not parse public key: %v", err)
+		return nil, fmt.Errorf("could not parse public key: %v", err)
 	}
 	ecdsaKey, ok := pubKeyDecoded.(*ecdsa.PublicKey)
 	if !ok {
-		return fmt.Errorf("public key is not ecdsa: %v", pubKeyDecoded)
+		return nil, fmt.Errorf("public key is not ecdsa: %v", pubKeyDecoded)
 	}
 
 	data, err := hex.DecodeString(*rekordEntry.RekordObj.Data.Hash.Value)
 	if err != nil {
-		return fmt.Errorf("could not decode hash of data: %v", rekordEntry.RekordObj.Data.Hash.Value)
+		return ecdsaKey, fmt.Errorf("could not decode hash of data: %v", rekordEntry.RekordObj.Data.Hash.Value)
```

Comment:
Please don't return the public key unless it was actually verified (also clarify this in the function comment); this is guaranteed to be misused, since there is no valid use for a key that did not actually sign the payload that it claims to have signed.

Reply:
But it's not "verifying the public key", it's verifying the signature against the public key...  but ok will do.

Reply:
fixed.